habitat-lab/habitat/tasks/rearrange/utils.py

```diff
@@ -431,7 +437,53 @@ def place_robot_at_closest_point(
             f"Object {target_position} is out of bounds but trying to set robot position to {agent_pos}"
         )
 
+    # Get the normal agent position
+    agent_pos = sim.safe_snap_point(target_position)
```

Comment:
Put all the added logic in the `if` statement for `navmesh_offset`. This code is called very often and there shouldn't be any unnecessary operations if `navmesh_offset` is None.